ddl/ddl_api.go

```diff
@@ -2789,7 +2795,7 @@ func checkTableDefCompatible(source *model.TableInfo, target *model.TableInfo) e
 	// Col compatible check
 	for i, sourceCol := range source.Cols() {
 		targetCol := target.Cols()[i]
```

Comment:
Note: `target.Cols()[i]` might be nil or out of slice range. These error will occur panic.